### PR TITLE
run test in dockerfile

### PR DESCRIPTION
**Pull request by akomm**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | yes
| Fixed tickets | #447
| License       | MIT

@mcg-web 

This includes how-to in CONTRIBUTING.md but I am not done. I will update this comment with some different proposals, because this can be done nicer in different ways. But the way it is currently already works for `test` and `code-quality`

- Move Dockerfile & .dockerignore to `some` subdirectory
- Simplify command: either 1. create a bash script which maps a command to composer in container **or** 2. add the command to composer scripts. The first would require only docker, the later would require docker, composer and php. Also composer command would with passing arguments look like this: `composer docker -- test -- --filter=SomeTestFilter`.
- Multi-Stage build so you can actually run test with different php versions (but not all at once if you do not want)
- While the most time consuming part of docker build is only required once per composer.json change and on initial build, there is still a bit of delay to run the build through all the cache layer. This can be improved by have docker build separate and not run on every test/code-quality command. However refreshing the build in the same command ensures you test updated source and do not forget to run docker build when you make changes. So in worst case what can happen is you forget to run docker build and run the test on old source. So in my opinion the saving of ~2-3 seconds not rebuilding (with cache) is not worth the bothering. Also its more convenient if you do not have to `docker build . -t graphql-test` everytime you make changes before you run any test.

### CODE REVIEW

**Comment by akomm** [comment]
@mcg-web if just picked some things initially. We might change it if you review :)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        php7-mbstring \
+        php7-openssl \
+        php7-phar \
+        php7-tokenizer \
+        php7-xml \
+        php7-xmlwriter \
+        php7-dom \
+    ;
+
+RUN set -eu; \
```
[the remark this answers is not in the document]
Comment (akomm) [reply Dockerfile]: Good idea!

**Comment by akomm** [comment]
Hm. Github commits one by one, cleaner to make a single commit manually and push.

**Comment by akomm** [comment]
I think the diff, the lines were not marked all as remove. :) Dockerfile is now messed up

**Comment by mcg-web** [comment]
Sorry but suggestions can be added only on one line :disappointed:

**Comment by akomm** [comment]
I normaly do it manually local, but got to test things what works better. 

Btw do you prefer full history or rebased/amended commits on WIP PRs?

**Comment by mcg-web** [comment]
full history when necessary otherwise rebased commit are just fine too.

**Pull request merged**